be/src/vec/functions/function_compress.cpp

```diff
@@ -0,0 +1,242 @@
+#include "vec/functions/function.h"
+#include "vec/functions/simple_function_factory.h"
+
+namespace doris {
+class FunctionContext;
+} // namespace doris
+
+namespace doris::vectorized {
+
+class FunctionCompress : public IFunction {
+    string hex_itoc = "0123456789ABCDEF";
+
+public:
+    static constexpr auto name = "compress";
+    static FunctionPtr create() { return std::make_shared<FunctionCompress>(); }
+
+    String get_name() const override { return name; }
+
+    size_t get_number_of_arguments() const override { return 1; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        return std::make_shared<DataTypeString>();
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        uint32_t result, size_t input_rows_count) const override {
+        // Get the compression algorithm object
+        BlockCompressionCodec* compression_codec;
+        RETURN_IF_ERROR(get_block_compression_codec(segment_v2::CompressionTypePB::ZLIB,
+                                                    &compression_codec));
+
+        const auto& arg_column =
+                assert_cast<const ColumnString&>(*block.get_by_position(arguments[0]).column);
+        auto result_column = ColumnString::create();
+
+        auto& arg_data = arg_column.get_chars();
+        auto& arg_offset = arg_column.get_offsets();
+        const char* arg_begin = reinterpret_cast<const char*>(arg_data.data());
+
+        auto& col_data = result_column->get_chars();
+        auto& col_offset = result_column->get_offsets();
+        col_offset.resize(input_rows_count);
+
+        faststring compressed_str;
+        Slice data;
+        for (size_t row = 0; row < input_rows_count; row++) {
+            size_t length = arg_offset[row] - arg_offset[row - 1];
+            data = Slice(arg_begin, length);
+
+            // Z_MEM_ERROR and Z_BUF_ERROR are already handled in compress, making sure st is always Z_OK
+            auto st = compression_codec->compress(data, &compressed_str);
+
+            size_t idx = col_data.size();
+            if (!length) { // data is ''
+                col_data.resize(col_data.size() + 2);
+                col_data[idx] = '0', col_data[idx + 1] = 'x';
+                col_offset[row] = col_offset[row - 1] + 2;
+                continue;
+            }
+
+            // first ten digits represent the length of the uncompressed string
+            col_data.resize(col_data.size() + 10);
+            col_data[idx] = '0', col_data[idx + 1] = 'x';
+            for (size_t i = 0; i < 4; i++) {
+                unsigned char byte = (length >> (i * 8)) & 0xFF;
+                col_data[idx + 2 + i * 2] = hex_itoc[byte >> 4]; // higher four
+                col_data[idx + 3 + i * 2] = hex_itoc[byte & 0x0F];
+            }
+            idx += 10;
+
+            col_data.resize(col_data.size() + 2 * compressed_str.size());
```

Comment:
like comment for `uncompress`, add a reserve operation before the for loop may be better.